### ISSUE TITLE
Files are flagged as ignored while they are not

### ISSUE THREAD

**Issue by adrienball** [hsz/idea-gitignore#566]
### Prerequisites

* [x] Plugin is in the latest version
* [x] Issue was not reported yet
* [ ] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

There is something wrong with .gitignore files in subdirectories when the .gitignore file contains:

```
*
!.gitignore
```

In such cases, it seems that all the files having a name starting with the name of the subdirectory will be considered as ignored by the plugin.

I created a toy example that reproduces this issue 100% of the time, cf Steps to Reproduce section.

### Steps to Reproduce

![image](https://user-images.githubusercontent.com/11029979/48545140-7f7e8700-e8c5-11e8-8496-e95204e79494.png)

**Expected behavior:**

The foobar.py and foobar_2.py are not flagged as ignored.

**Actual behavior:** 

They are flagged as ignored.

**Reproduces how often:** 

100%

### Versions

**Plugin:** 
```
Version: 3.0.0.182
```

**IDE:** 

```
PyCharm 2018.2.5 (Community Edition)
Build #PC-182.5107.22, built on November 13, 2018
JRE: 1.8.0_152-release-1248-b22 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.12.6
```

**OS:** 

```
macOS 10.12.6
```

**Comment by hsz** [comment]
@adrienball Thank you for reporting this issue. I've fixed it and it'll be released with v3.1.0

**Comment by hsz** [comment]
Fixed and released with v3.1.1
